systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java

```diff
@@ -122,21 +120,16 @@ void testDeployUnsupportedKafka() {
         LOGGER.info("Wait until Zookeeper stateful set is ready");
         StUtils.waitForAllStatefulSetPodsReady(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME), 1);
 
-        LOGGER.info("Find warning about Kafka version in Cluster Operator log");
-        String expectedLog = "createOrUpdate failed. Version 6.6.6 is not supported. Supported versions are:";
-        String coPodName = kubeClient().getClusterOperatorPodName();
-        StUtils.waitUntilMessageIsInLogs(coPodName, "strimzi-cluster-operator", expectedLog);
-        LOGGER.info("Warning is present in Cluster Operator log, going to teardown Created Kafka");
+        StUtils.waitUntilKafkaStatusConditionIsPresent(CLUSTER_NAME);
 
-        assertThat(kubeClient().logs(coPodName), not(containsString("NullPointer")));
-        assertKafkaStatus("NotReady", "Version 6.6.6 is not supported");
-    }
+        Condition condition = testMethodResources().kafka().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus().getConditions().get(0);
 
-    // This should be moved to StUtils in the futures
-    void assertKafkaStatus(String status, String message) {
-        KafkaStatus kafkaStatus = testMethodResources().kafka().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus();
-        assertThat("Kafka Kafka cluster status should be " + status, kafkaStatus.getConditions().get(0).getType(), is(status));
-        assertThat("Kafka cluster status message should be different", kafkaStatus.getConditions().get(0).getMessage(), containsString(message));
+        verifyCRStatusCondition(
+                condition,
+                "Version 6.6.6 is not supported. Supported versions are: 2.2.1, 2.3.0.",
```

Comment:
This is very fragile. Would it be better to just check that _"Version 6.6.6 is not supported_ is part of that message?